CspSubmissionPlugin.inc.php

```diff
@@ -904,84 +844,70 @@ function TemplateManager_fetch($hookName, $args) {
 
-			$args[4] = $templateMgr->fetch($this->getTemplateResource('queryForm.tpl'));
+			/// Se o usuário for o autor, o destinatário será a secretaria, caso contrário, o destinatário será o autor
+			$submissionDAO = Application::getSubmissionDAO();
+			$submission = $submissionDAO->getById($request->getUserVar('submissionId'));
+			$publication = $submission->getCurrentPublication();
+			$userDao = DAORegistry::getDAO('UserDAO');
+			$authorDao = DAORegistry::getDAO('AuthorDAO');
+			$author = $authorDao->getById($publication->getData('primaryContactId'));
+			$author = $userDao->getUserByEmail($author->getData('email'));
 
-			return true;
+			if ($author->getData('id') == $_SESSION["userId"]) {
 
-		}elseif ($args[1] == 'controllers/grid/queries/form/queryForm.tpl' && $stageId == "4") {
+				$mail = new MailTemplate('MSG_AUTOR_SECRETARIA');
+				$templateSubject['MSG_AUTOR_SECRETARIA'] = $mail->_data["subject"];
+				$templateBody['MSG_AUTOR_SECRETARIA'] = $mail->_data["body"];
 
-			$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
-			$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO'); /* @var $stageAssignmentDao StageAssignmentDAO */
-			$manager = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_MANAGER);
-			$assistent = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_ASSISTANT);
-			$stageAssignmentsFactory = $stageAssignmentDao->getBySubmissionAndStageId($request->getUserVar('submissionId'), null, null, $_SESSION["userId"]);
+				$templateMgr->assign('author', true);
+				$templateMgr->assign('to', 14); // BUSCAR ID DA SECRETARIA
+				$templateMgr->assign('toName', 'Secretaria');
+				$templateMgr->assign('from', $_SESSION["userId"]);
 
-			while ($stageAssignment = $stageAssignmentsFactory->next()) {
-				if (in_array($stageAssignment->getUserGroupId(), $manager)) {
-					$isManager = true;
+			}else{
+				if($stageId == "1"){ // Se o estágio for Pré-avaliação, template específico é exibido
+					$mail = new MailTemplate('PRE_AVALIACAO');
+					$templateSubject['PRE_AVALIACAO'] = $mail->_data["subject"];
+					$templateBody['PRE_AVALIACAO'] = $mail->_data["body"];
 				}
-				if (in_array($stageAssignment->getUserGroupId(), $assistent)) {
-					$isAssistent = true;
+				if($stageId == "4"){ // Se o estágio for Edição de texto, templates específicos são exibidos para cada perfil
+					$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
+					$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO'); /* @var $stageAssignmentDao StageAssignmentDAO */
+					$manager = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_MANAGER);
+					$assistent = $userGroupDao->getUserGroupIdsByRoleId(ROLE_ID_ASSISTANT);
+					$stageAssignmentsFactory = $stageAssignmentDao->getBySubmissionAndStageId($request->getUserVar('submissionId'), null, null, $_SESSION["userId"]);
+
+					while ($stageAssignment = $stageAssignmentsFactory->next()) {
```

Comment:
dado o if que tem abaixo, eu entendi que só um deles pode ser true certo? daria pra adicionar um if no começo do while pra parar assim que encontrar um deles
```suggestion
					$isManager = false;
					$isAssistent = false;
					while ($stageAssignment = $stageAssignmentsFactory->next()) {
						if ($isManager || $isAssistent){
							break;
						}
```

ou adicionar essa regra no proprio while, `while ($stageAssignment = $stageAssignmentsFactory->next() && !$isManager && !$isAssistent ) {`